### PR TITLE
chore: add hilla-dev-mode in vaadin-dev

### PR DESCRIPTION
**Pull request by tltv**
Fixes: #4946

Note! Need to merge first Hilla's https://github.com/vaadin/hilla/pull/2001.

### CODE REVIEW

**Review by ZheSun88 — APPROVED** [review]

File: vaadin-dev/pom.xml [hunk 1]
```diff
@@ -44,6 +44,11 @@
             <artifactId>vaadin-dev-bundle</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.vaadin.hilla</groupId>
+            <artifactId>hilla-dev-mode</artifactId>
+        </dependency>
+
```
Comment (manolo) [inline vaadin-dev/pom.xml:51]: why is this needed here instead of on the spring starter ?

**Review by manolo — CHANGES_REQUESTED** [review]

File: vaadin-dev/pom.xml (hunk 1 above)
Comment (tltv) [reply vaadin-dev/pom.xml:51]: Discussed in slack. Conclusion: remove it from spring starter.

File: vaadin-platform-test/src/test/java/com/vaadin/platform/test/ChromeComponentsIT.java [hunk 2]
```diff
@@ -83,8 +83,13 @@ protected String getTestPath() {
     }
 
     @Before
-    public void load(){
-        $(NotificationElement.class).waitForFirst();
+    public void load() {
+        if (getDriver().getPageSource().contains("<div class=\"message\">Building front-end development bundle</div>")) {
+            // Wait for the development bundle to be loaded
+            $(NotificationElement.class).waitForFirst(180);
```
Comment (manolo) [inline vaadin-platform-test/src/test/java/com/vaadin/platform/test/ChromeComponentsIT.java:89]: I feel that ff the problem is that we need for the bundle to be compiled, probably this fix should be in the testbench instead

File: scripts/generator/templates/template-prod-bundle-pom.xml [hunk 3]
```diff
@@ -32,6 +32,11 @@
             <version>${project.version}</version>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>com.vaadin.hilla</groupId>
+            <artifactId>hilla-dev-mode</artifactId>
```
Comment (mshabarov) [inline scripts/generator/templates/template-prod-bundle-pom.xml:37]: This shouldn't go into prod pre-compiled bundle, as it brings HotSwapAgent plugin and vaadin-dev-server

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: scripts/generator/templates/template-prod-bundle-pom.xml (hunk 3 above)
Comment (tltv) [reply scripts/generator/templates/template-prod-bundle-pom.xml:37]: I agree. And hilla-dev-mode should be excluded from production builds in general. This is not automatic however but build configs can exclude whole vaadin-dev from the production build.

**Review by manolo — APPROVED** [review]
